cli/scancommands.go

```diff
@@ -407,6 +410,27 @@ func AuditCmd(c *components.Context) error {
 		auditCmd.SetScansToPerform(subScans)
 	}
 
+	// Validate that there is a sast scan before setting the sast rules
+	if c.GetStringFlagValue(flags.SastRules) != "" {
+		if !slices.Contains(auditCmd.ScansToPerform(), utils.SubScanType(flags.Sast)) {
+			return pluginsCommon.PrintHelpAndReturnError(fmt.Sprintf("flag '--%s' can only be used with '--%s'", flags.SastRules, flags.Sast), c)
+		}
+
+		sastRules := c.GetStringFlagValue(flags.SastRules)
+		if _, err := os.Stat(sastRules); err != nil {
+			switch {
+			case errors.Is(err, fs.ErrNotExist):
+				return pluginsCommon.PrintHelpAndReturnError(fmt.Sprintf("file '%s' does not exist", sastRules), c)
+			case errors.Is(err, fs.ErrPermission):
+				return pluginsCommon.PrintHelpAndReturnError(fmt.Sprintf("permission denied to read sast rules file '%s': %v", sastRules, err), c)
+			default:
+				return pluginsCommon.PrintHelpAndReturnError(fmt.Sprintf("error reading sast rules file '%s': %v", sastRules, err), c)
+			}
+		}
+
+		auditCmd.SetSastRules(sastRules)
+	}
+
```

Comment:
```suggestion
	// Validate that there is a sast scan before setting the sast rules
	sastRulesFile := c.GetStringFlagValue(flags.SastRules)
	if sastRulesFile != "" {
	   // Check if file exists
	   if exists, err := fileutils.IsFileExists(cdxFilePath, false); err != nil {	  
	       return pluginsCommon.PrintHelpAndReturnError(fmt.Sprintf("file '%s' does not exist: %s", sastRules, err.Error()), c)
	   }
	    // Validate scan is performed
	   if len(params.ScansToPerform()) > 0 && !slices.Contains(params.ScansToPerform(), utils.SecretsScan)    {
	       return pluginsCommon.PrintHelpAndReturnError(fmt.Sprintf("flag '--%s' can only be used with '--%s'", flags.SastRules, flags.Sast), c)
	   }
		auditCmd.SetSastRules(sastRulesFile)
	}

```

try to reduce complexity and duplicate calls to funcs. use utils in repo